Assets/_BForBoss/_Weapons/Scripts/Weapons/Weapon.cs

```diff
@@ -15,13 +15,15 @@ public class Weapon
          private float _elapsedRateOfFire;
          private float _elapsedReloadDuration;
          private int _ammunitionAmount;
+         private bool _isActive = false;
```

Comment:
There was a bug where swapping weapons has a high chance of shooting previous weapon. This is so make sure it can't be shot when `gameobject.setActive(false)`